viewing_party/party.py

```diff
@@ -1,23 +1,102 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    else:
+        return None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+                #the movie_title is present in watchlist
+                user_data["watchlist"].remove(movie)
+                add_to_watched(user_data, movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    if(len(user_data["watched"]) == 0):
+        return 0
+    sum = 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    average = sum / len(user_data["watched"])
+    return average
+
+def get_most_watched_genre(user_data):
+    if (len(user_data["watched"]) == 0):
+        return None
+    most_watched = ""
+    genre_list = []
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+    most_watched = max(set(genre_list), key = genre_list.count)
+    return most_watched
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_friends_watched(user_data):

```

Comment:
Could sets be used instead of some of the nested loops in your functions?

Reply:
On hindsight, perhaps. especially since there is a lot of "in", "not in" code.